nipype/utils/draw_gantt_chart.py

```diff
@@ -0,0 +1,585 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+"""Module to draw an html gantt chart from logfile produced by
+callback_log.log_nodes_cb()
+"""
+
+# Import packages
+import json
+from dateutil import parser
+import datetime
+import random
+from collections import OrderedDict
+# Pandas
+try:
+    import pandas as pd
+except ImportError:
+    print('Pandas not found; in order for full functionality of this module '\
+          'install the pandas package')
+    pass
+
+def create_event_dict(start_time, nodes_list):
+    '''
+    Function to generate a dictionary of event (start/finish) nodes
+    from the nodes list
+
+    Parameters
+    ----------
+    start_time : datetime.datetime
+        a datetime object of the pipeline start time
+    nodes_list : list
+        a list of the node dictionaries that were run in the pipeline
+
+    Returns
+    -------
+    events : dictionary
+        a dictionary where the key is the timedelta from the start of
+        the pipeline execution to the value node it accompanies
+    '''
+
+    # Import packages
+    import copy
+
+    events = {}
+    for node in nodes_list:
+        # Format node fields
+        try:
+            estimated_threads = float(node['num_threads'])
+        except:
+            estimated_threads = 1
+        try:
+            estimated_memory_gb = float(node['estimated_memory_gb'])
+        except:
+            estimated_memory_gb = 1.0
+        try:
+            runtime_threads = float(node['runtime_threads'])
+        except:
+            runtime_threads = 0
+        try:
+            runtime_memory_gb = float(node['runtime_memory_gb'])
+        except:
```

Comment:
Actually each of the above can be replaced, byt a .get() with a default value http://stackoverflow.com/a/9359011